### PR TITLE
Refs #12634 - added Grub templates for UEFI

### PR DESCRIPTION
**Pull request by lzap**
This is for PXE UEFI support:

https://github.com/theforeman/foreman/pull/3679

Templates that needs to be provided in this patch:
- [x] PXEGrub2 default local boot
- [x] PXEGrub global default
- [x] PXEGrub2 global default
- [x] Kickstart default PXEGrub
- [x] Kickstart default PXEGrub2
- [ ] Jumpstart default PXEGrub
- [ ] Jumpstart default PXEGrub2
- [x] pxegrub_discovery
- [x] pxegrub2_discovery

### CODE REVIEW

**Comment by lzap** [comment]
@timogoebel I am going to put this in the Foreman Core patch, can you check it please?

**Comment by lzap** [comment]
Wait, I actually need to add PXEGrub2 _and_ PXEGrub OS templates, I will provide ones for _Red Hat_ as well as _Ubuntu/Debian_ distributions.

**Comment by lzap** [comment]
(Edit: Moved the summary to the description)

To test this patch, you can use templates plugin:

```
be rake templates:sync prefix= verbose=1 repo="https://github.com/lzap/community-templates" branch="pxegrub2"
```

**Comment by timogoebel** [comment]
@lzap : The list sounds about right. `PXEGrub2_local` looks good. Couldn't really test it as my VMs don't have a partition layout yet and Grub 2 complains, that it does't know the disk "(hd0,0)".

**Comment by lzap** [comment]
Pushed all templates, these are untested for now, will start real testing (actual provisioning) tomorrow.

**Comment by lzap** [comment]
Just for the record, do not merge until the core PR is merged, then I will update this to reflect the change.

File: pxe/PXEGrub_local.erb [hunk 1]
```diff
@@ -2,11 +2,11 @@
 kind: PXEGrub
 name: PXEGrub default local boot
 %>
-
-<%# This template has special name (do not change it) and it is used for booting already provisioned hosts. %>
+<%# Used to boot provisioned hosts, do not associate or change the name. %>
```
Comment (dLobatog) [inline pxe/PXEGrub_local.erb:5]: We can enforce that in Foreman by locking it by default

Comment (lzap) [inline pxe/PXEGrub_local.erb:5]: > We can enforce that in Foreman by locking it by default

Added to the PR:

```
    { :name => 'PXELinux global default', :source => 'pxe/PXELinux_default.erb', :template_kind => kinds[:PXELinux], :locked => true },
    { :name => 'PXEGrub global default', :source => 'pxe/PXEGrub_default.erb', :template_kind => kinds[:PXEGrub], :locked => true },
    { :name => 'PXEGrub2 global default', :source => 'pxe/PXEGrub2_default.erb', :template_kind => kinds[:PXEGrub2], :locked => true },
    { :name => 'PXELinux default local boot', :source => 'pxe/PXELinux_local.erb', :template_kind => kinds[:PXELinux], :locked => true },
    { :name => 'PXEGrub default local boot', :source => 'pxe/PXEGrub_local.erb', :template_kind => kinds[:PXEGrub], :locked => true },
    { :name => 'PXEGrub2 default local boot', :source => 'pxe/PXEGrub2_local.erb', :template_kind => kinds[:PXEGrub2], :locked => true },
```

## 

Later,
 Lukas #lzap Zapletal

**Comment by lzap** [comment]
Merged since this was merged into core already and sync would overwrite the changes.

**Pull request merged**